### ISSUE TITLE
setErrors not working correctly when updating to angular 13 and using the prerelease

### ISSUE THREAD

**Issue by jcoble** [ngx-formly/ngx-formly#3123]
**Description**

Hello,

Just wanted to say I'm loving formly.  I created a whole drag and drop page for our users to extend the forms we've already created.  I am porting over a massive winforms enterprise application that is 100's of forms big, so this will also save time.

When I updated to V13 of angular and also using the latest prerelease v6.0.0-next.6, when using setErrors to display server errors after submit, it does not display the error.  It only will, once I actually go physically touch (focus) the input.  I've tried everything over the last couple days and was just thinking I was doing something wrong, until I I used your stackblitz example, worked, and then updated, and then did not.

Even before submit if manually setting an error, it actually will show the error the first time.  Then go focus and type in the input, the error is gone.  Then setErrors again on same field and will not display until focused physically.  markAsTouched or any other way i tried will not display that error until focused by ui.

I am using the material ui package.  I'm guessing it has something to do with the errorStateMatcher but not sure.  Pretty new to front end web dev.

**Minimal Reproduction**

[https://stackblitz.com/edit/ngx-formly-ui-material-pxk7en?file=src/app/app.component.ts](url)

Enter something in the first input.
Hit Set Errors button
focus input and type again so error is gone
Hit Set Errors again and nothing will show
focus input and error appears

Or fill out form
Hit Submit
Hit Set Errors and nothing shows
focus input and error appears

Thanks for the help!

-Jesse

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v6.0.0-next.7](https://github.com/ngx-formly/ngx-formly/releases/tag/v6.0.0-next.7) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!

**Comment by Eisie96** [comment]
We encountered another problem. On our register page we have multiple required formly fields. If we want to submit the form  without any given informations, the expectasion is, that all required fields show a custom error message. With the version formly v6.0.0-next.7 only the address fields work right.

![image](https://user-images.githubusercontent.com/44707835/151129027-6c4e613e-e6be-442c-b6da-0a77de9f2e58.png)

After some inspectations we found out, that showError() is called just for the address-fields after submit. The error message shows up, if the not working fields are clicked.

Here is the configuration for the formly fields:

```json
[
  {
      "type": "ish-registration-heading-field",
      "templateOptions": {
          "headingSize": "h2",
          "heading": "account.register.personal_information.heading",
          "showRequiredInfo": true,
          "disabled": false
      },
      "id": "formly_11_ish-registration-heading-field__3",
      "hooks": {},
      "modelOptions": {},
      "validation": {
          "messages": {}
      },
      "resetOnHide": true,
      "wrappers": [],
      "expressionProperties": {}
  },
  {
      "type": "ish-fieldset-field",
      "templateOptions": {
          "fieldsetClass": "row",
          "childClass": "col-md-10 col-lg-8 col-xl-6"
      },
      "fieldGroup": [
          {
              "key": "firstName",
              "type": "ish-text-input-field",
              "templateOptions": {
                  "label": "account.default_address.firstname.label",
                  "required": true,
                  "placeholder": "",
                  "disabled": false,
                  "type": "text"
              },
              "validators": {
                  "validation": [
                      null
                  ]
              },
              "validation": {
                  "messages": {
                      "required": "account.address.firstname.missing.error",
                      "noSpecialChars": "account.name.error.forbidden.chars"
                  }
              },
              "id": "formly_14_ish-text-input-field_firstName_0",
              "hooks": {},
              "modelOptions": {},
              "resetOnHide": true,
              "wrappers": [
                  "form-field-horizontal",
                  "validation"
              ],
              "expressionProperties": {}
          },
          {
              "key": "lastName",
              "type": "ish-text-input-field",
              "templateOptions": {
                  "label": "account.default_address.lastname.label",
                  "required": true,
                  "placeholder": "",
                  "disabled": false,
                  "type": "text"
              },
              "validators": {
                  "validation": [
                      null
                  ]
              },
              "validation": {
                  "messages": {
                      "required": "account.address.lastname.missing.error",
                      "noSpecialChars": "account.name.error.forbidden.chars"
                  }
              },
              "id": "formly_14_ish-text-input-field_lastName_1",
              "hooks": {},
              "modelOptions": {},
              "resetOnHide": true,
              "wrappers": [
                  "form-field-horizontal",
                  "validation"
              ],
              "expressionProperties": {}
          },
          {
              "key": "phoneHome",
              "type": "ish-phone-field",
              "templateOptions": {
                  "label": "account.profile.phone.label",
                  "required": false,
                  "placeholder": "",
                  "disabled": false,
                  "attributes": {
                      "maxlength": 20
                  },
                  "type": "tel"
              },
              "id": "formly_14_ish-phone-field_phoneHome_2",
              "hooks": {},
              "modelOptions": {},
              "validation": {
                  "messages": {
                      "phone": "form.phone.error.invalid",
                      "required": "form.phone.error.required"
                  }
              },
              "resetOnHide": true,
              "validators": {},
              "wrappers": [
                  "form-field-horizontal",
                  "validation"
              ],
              "expressionProperties": {}
          }
      ],
      "id": "formly_11_ish-fieldset-field__4",
      "hooks": {},
      "modelOptions": {},
      "validation": {
          "messages": {}
      },
      "resetOnHide": true,
      "wrappers": [],
      "expressionProperties": {}
  },
  {
      "type": "ish-registration-heading-field",
      "templateOptions": {
          "headingSize": "h2",
          "heading": "account.register.address.headding",
          "subheading": "account.register.address.message",
          "showRequiredInfo": true,
          "disabled": false
      },
      "id": "formly_14_ish-registration-heading-field__5",
      "hooks": {},
      "modelOptions": {},
      "validation": {
          "messages": {}
      },
      "resetOnHide": true,
      "wrappers": [],
      "expressionProperties": {}
  },
  {
      "type": "ish-fieldset-field",
      "templateOptions": {
          "fieldsetClass": "row",
          "childClass": "col-md-10 col-lg-8 col-xl-6"
      },
      "fieldGroup": [
          {
              "type": "ish-registration-address-field",
              "templateOptions": {
                  "businessCustomer": false,
                  "disabled": false
              },
              "id": "formly_17_ish-registration-address-field__0",
              "hooks": {},
              "modelOptions": {},
              "validation": {
                  "messages": {}
              },
              "resetOnHide": true,
              "wrappers": [],
              "expressionProperties": {}
          },
          {
              "type": "ish-registration-tac-field",
              "key": "termsAndConditions",
              "templateOptions": {
                  "required": true,
                  "label": "",
                  "placeholder": "",
                  "disabled": false
              },
              "validators": {
                  "validation": [
                      null
                  ]
              },
              "id": "formly_17_ish-registration-tac-field_termsAndConditions_1",
              "hooks": {},
              "modelOptions": {},
              "validation": {
                  "messages": {}
              },
              "resetOnHide": true,
              "wrappers": [],
              "expressionProperties": {}
          },
          {
              "type": "ish-captcha-field",
              "templateOptions": {
                  "topic": "register",
                  "disabled": false
              },
              "id": "formly_17_ish-captcha-field__2",
              "hooks": {},
              "modelOptions": {},
              "validation": {
                  "messages": {}
              },
              "resetOnHide": true,
              "wrappers": [],
              "expressionProperties": {}
          }
      ],
      "id": "formly_14_ish-fieldset-field__6",
      "hooks": {},
      "modelOptions": {},
      "validation": {
          "messages": {}
      },
      "resetOnHide": true,
      "wrappers": [],
      "expressionProperties": {}
  }
]
```

Thanks for help!
